ext/spl/spl_fixedarray.c

```diff
@@ -309,24 +309,56 @@ static zend_object *spl_fixedarray_object_clone(zend_object *old_object)
 	return new_object;
 }
 
+static zend_long spl_offset_convert_to_long(zval *offset) /* {{{ */
+{
+	try_again:
+	switch (Z_TYPE_P(offset)) {
+		case IS_STRING: {
+			zend_ulong index;
+			if (ZEND_HANDLE_NUMERIC(Z_STR_P(offset), index)) {
+				return (zend_long) index;
+			}
+			break;
+		}
+		case IS_DOUBLE:
+			return zend_dval_to_lval_safe(Z_DVAL_P(offset));
+		case IS_LONG:
+			return Z_LVAL_P(offset);
+		case IS_FALSE:
+			return 0;
+		case IS_TRUE:
+			return 1;
+		case IS_REFERENCE:
+			offset = Z_REFVAL_P(offset);
+			goto try_again;
+		case IS_RESOURCE:
+			zend_error(E_WARNING, "Resource ID#%d used as offset, casting to integer (%d)",
+				Z_RES_HANDLE_P(offset), Z_RES_HANDLE_P(offset));
+			return Z_RES_HANDLE_P(offset);
+	}
+
+	zend_type_error("Illegal offset type");
```

Comment:
Would including the offset type make sense here - it may make this error easier to debug

```c
zend_type_error("Illegal offset type %s", zend_zval_type_name(offset);
```

It's used in a few places

```c
php-src/ext/session/session.c
2515:                                           zend_type_error("%s(): Option \"%s\" must be of type string|int|bool, %s given",
2516-                                                   get_active_function_name(), ZSTR_VAL(str_idx), zend_zval_type_name(value)
2517-                                           );
```